src/foam/nanos/medusa/ClusterConfigReplayDAO.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.medusa',
+  name: 'ClusterConfigReplayDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `On status change to ONLINE initiate replay`,
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.dao.Sink',
+    'static foam.mlang.MLang.AND',
+    'static foam.mlang.MLang.EQ',
+    'static foam.mlang.MLang.GTE',
+    'static foam.mlang.MLang.LT',
+    'static foam.mlang.MLang.MAX',
+    'static foam.mlang.MLang.MIN',
+    'static foam.mlang.MLang.OR',
+    'foam.mlang.sink.Count',
+    'foam.mlang.sink.Max',
+    'foam.mlang.sink.Min',
+    'foam.mlang.sink.Sequence',
+    'foam.nanos.logger.PrefixLogger',
+    'foam.nanos.logger.Logger'
+  ],
+
+  properties: [
+    {
+      name: 'logger',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.logger.Logger',
+      visibility: 'HIDDEN',
+      transient: true,
+      javaFactory: `
+        return new PrefixLogger(new Object[] {
+          this.getClass().getSimpleName()
+        }, (Logger) getX().get("logger"));
+      `
+    }
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+      ClusterConfig nu = (ClusterConfig) obj;
+      ClusterConfig old = (ClusterConfig) find_(x, nu.getId());
+      nu = (ClusterConfig) getDelegate().put_(x, nu);
+      if ( old != null &&
+           old.getStatus() != nu.getStatus() &&
+           nu.getStatus() == Status.ONLINE ) {
+
+        getLogger().info(nu.getName(), old.getStatus().getLabel(), "->", nu.getStatus().getLabel().toUpperCase());
+
+        ClusterConfig config = nu;
+        ClusterConfigSupport support = (ClusterConfigSupport) x.get("clusterConfigSupport");
+        ClusterConfig myConfig = support.getConfig(x, support.getConfigId());
+
+        if ( support.getStandAlone() &&
+             nu.getType() == MedusaType.NODE ) {
+          getLogger().debug("standalone");
+          ReplayingInfo replaying = (ReplayingInfo) x.get("replayingInfo");
+          replaying.setStartTime(new java.util.Date());
+
+          DAO dao = (DAO) x.get("medusaNodeDAO");
+          Min min = (Min) MIN(MedusaEntry.INDEX);
```

Comment:
min is unused?

Reply:
Yes.  The incomplete intent is make a more focused replay request to the nodes. Presently max is ignored as well, and the nodes just replay everything.